lib/rails_event_store/repositories/event_repository.rb

```diff
@@ -89,12 +89,13 @@ def read_all_streams_backward(start_event_id, count)
 
       def build_event_entity(record)
         return nil unless record
-        ::RailsEventStore::Event.new(record.data.merge(
-          event_type: record.event_type,
-          event_id:   record.event_id,
-          metadata:   record.metadata))
+        record.event_type.constantize.new(
```

Comment:
I would prefer move this code to `ruby_event_store`. It should not be Repository who is responsible for recreating an event, here it should be something like:

```ruby
def build_event_entity(record)
   return nil unless record
   event_factory.call(record.data.merge(
     event_id: record.event_id,
     metadata: record.metadata
   ))
end
```